packages/flutter_tools/lib/src/commands/format.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:path/path.dart' as path;
+
+import '../base/process.dart';
+import '../cache.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+
+class FormatCommand extends FlutterCommand {
+  @override
+  final String name = 'format';
+
+  @override
+  final String description = 'Format one or more dart files.';
+
+  @override
+  bool get requiresProjectRoot => false;
+
+  @override
+  String get invocation => "${runner.executableName} $name <one or more paths>";
+
+  @override
+  Future<int> runInProject() async {
+    if (argResults.rest.isEmpty) {
```

Comment:
We could also choose to improve the UI of dartfmt a bit. If you're in a project - and no other args are given - we could recurse into source directories and format those files. This would operate similar to how flutter analyze works.

I don't think we need to do that in this PR, but something to think about.